tests/bluetooth/bsim/host/adv/encrypted/css_sample_data/src/central.c

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+extern const uint8_t *session_key;
+extern const uint8_t *iv;
+extern uint8_t size_sample_ad_data;
+extern const uint8_t *sample_ad_data;
+extern uint8_t size_sample_ead;
+extern const uint8_t *sample_ead;
+
+extern int data_set;
+
+static bool data_parse_cb(struct bt_data *data, void *user_data)
+{
+	if (data->type == BT_DATA_ENCRYPTED_AD_DATA) {
+		int err;
+		size_t decrypted_data_size = BT_EAD_DECRYPTED_PAYLOAD_SIZE(data->data_len);
+
+		if (decrypted_data_size != size_sample_ad_data) {
+			LOG_ERR("Size of decrypted data: %d", decrypted_data_size);
+			LOG_ERR("Size of sample data: %d", size_sample_ad_data);
+			FAIL("Computed size of data does not match the size of the data from the "
+			     "sample. (data set %d)\n",
+			     data_set);
+		}
+
+		struct net_buf_simple *decrypted_buf = k_malloc(sizeof(struct net_buf_simple));
+
+		decrypted_buf->size = decrypted_data_size;
+		decrypted_buf->__buf =
+			k_malloc(sizeof(decrypted_buf->data[0]) * decrypted_data_size);
+		net_buf_simple_reset(decrypted_buf);
+
+		err = bt_ead_decrypt(session_key, iv, data->data, data->data_len,
+				     decrypted_buf->data);
+
+		if (err != 0) {
+			FAIL("Error during decryption.\n");
+		} else if (memcmp(decrypted_buf->data, sample_ad_data_1, decrypted_data_size)) {
+			LOG_HEXDUMP_ERR(decrypted_buf->data, decrypted_data_size,
+					"Decrypted data from bt_ead_decrypt:");
+			LOG_HEXDUMP_ERR(sample_ad_data_1, size_sample_ad_data,
+					"Expected data from sample:");
+			FAIL("Decrypted AD data does not match expected sample data. (data set "
+			     "%d)\n",
+			     data_set);
+		}
+
+		decrypted_buf->len = decrypted_data_size;
+
+		LOG_HEXDUMP_DBG(decrypted_buf->data, decrypted_data_size, "Raw decrypted data: ");
+
+		bt_data_parse(decrypted_buf, &data_parse_cb, NULL);
+
+		k_free(decrypted_buf->__buf);
+		k_free(decrypted_buf);
+
+		PASS("Central test passed. (data set %d)\n", data_set);
```

Comment:
Since we are passing this test in this case, we could return `false` to stop parsing any more AD structures